libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -128,7 +128,7 @@ typedef int ( * pfnMbedTlsSign )( void * ctx,
  * @brief Indicates that no PKCS #11 operation is underway for given session.
  *
  */
-#define pkcs11NO_OPERATION     ( ( CK_MECHANISM_TYPE ) 0xFFFFFFFFF )
+#define pkcs11NO_OPERATION     ( ( CK_MECHANISM_TYPE ) 0xFFFFul )
 
```

Comment:
Richard pointed this one out to me, it has 9 "f"'s and now it has 4, that is 36 bits vs 16 bits. It this correct? Should the value be a ul with only the bottom 16 bits assigined? Seems odd to me. Most of the time something like this is best done doing  (type)(-1) or similar ...

Reply:
Oops, this should be 8F's

Reply:
Applied the -1 suggestion

Reply:
That makes more sense then !

Reply:
The issue I had here is that unsigned long is not necessarily 32 bits, if it is more this is not the same as -1, if you cast -1 to the type it will always be as many F's as you can fit into the variable ...

Reply:
Wouldn't this cause compiler warnings of signedness conversion in cast?

Reply:
I tested with gcc-7 and interestingly, it does not raise a warning even with ` -Wall -Wextra -Wpedantic`  flags passed.